### DISCUSSION TITLE
Teensy 3.6 SPI for TFT+touch - change SPI clock pin?

### DISCUSSION THREAD

**Discussion by u1550l** [ImpulseAdventure/GUIslice#433] [Q&A] [answered]
Hello,
I am using GUIslice on a 2.8" Display 320*240 with Touch (http://www.lcdwiki.com/2.8inch_SPI_M...41_SKU:MSP2807) attached to a Teensy 3.6.
Display and touch are working great, but: 
Unfortunately SPI SCK0 shares the pin with the onboard LED, but I want to keep the led for my purposes.
According to Teensy's documentation there is a way use redirect the clock pin from 13 to 14 or 27
Who can tell me how to do this?

Thanks for your help

**Comment by Pconti31** [comment, marked as answer]
@u1550l I would try and edit your config file, say its teensy-adagfx-ili9341_t3-xpt2046.h
and change
```
//  #define ADAGFX_PIN_CLK      13
  #define ADAGFX_PIN_CLK      14
```
and somewhere in your app before you try to use the led 
```
pinMode(13,OUTPUT);
```
If that fails you may also need to hack GUIslice_drv_adagfx.cpp gslc_DrvInit() function to use:
```
  SPI.begin();
  SPI.setBitOrder(MSBFIRST); 
  SPI.setDataMode(SPI_MODE1);
  
// It is VERY IMPORTANT that the pins be reassigned AFTER ALL
// SPI initialization INCLUDING datamode or bitorder etc.

  // First reassign pin 13 to Alt1 so that it is not SCK but the LED still works
  CORE_PIN13_CONFIG = PORT_PCR_MUX(1);
  // and then reassign pin 14 to SCK
  CORE_PIN14_CONFIG = PORT_PCR_DSE | PORT_PCR_MUX(2);```


Paul--

**Reply by u1550l** [reply]
I can post the answer to my own question:
I got a reply from the developer of the Teensy:
 just call 
`SPI.setSCK(14);`
in setup() before any GUIslice code is called first.
This setSCK() is possible before SPI is accessed first.
